File: server/internal/core/configs.go
package core

import "time"

type Configs struct {
	Levels []LevelConfig `json:"levels"`
	Energy EnergyConfig  `json:"energy"`
}

type EnergyConfig struct {
	MaxEnergy               int `json:"maxEnergy"`
	RechargeIntervalSeconds int `json:"rechargeIntervalSeconds"`
}

func (c *EnergyConfig) RechargeInterval() time.Duration {
	return time.Duration(c.RechargeIntervalSeconds) * time.Second
}

type LevelConfig struct {
	EnergyCost   int `json:"energyCost"`
	MaxRolls     int `json:"maxRolls"`
	TargetNumber int `json:"targetNumber"`
	EnergyReward int `json:"energyReward"`
}

File: Assets/Scripts/Core/Runtime/Configs.cs
using System;

[Serializable]
public struct Configs
{
    public LevelConfig[] Levels;
    public EnergyConfig Energy;
}

[Serializable]
public struct EnergyConfig
{
    public int MaxEnergy;
    public TimeSpan RechargeInterval;
}

[Serializable]
public struct LevelConfig
{
    public int EnergyCost;
    public int MaxRolls;
    public int TargetNumber;
    public int EnergyReward;
}